src/main/java/org/opentripplanner/standalone/OTPMain.java

```diff
@@ -113,6 +113,14 @@ private static void startOTPServer(CommandLineParameters cli) {
     var loadApp = new LoadApplication(cli);
     var config = loadApp.config();
 
+    // optionally check if the config is valid and if not abort the startup process
+    if (cli.configCheck && config.hasInvalidProperties()) {
+      throw new OtpAppException(
+        "Configuration contains invalid properties (see above for details). " +
+        "Please fix your configuration or remove --configCheck from your OTP CLI parameters."
+      );
+    }
```

Comment:
Please extract this into a private method, and put the comment as JavaDoc.

Reply:
bd702cda23caf0b09aa342c932268f955a11469b